zen_queries/rest_framework.py

```diff
@@ -32,7 +33,9 @@ def get_serializer(self, *args, **kwargs):
         )
         if self.request.method == "GET":
             serializer = disable_serializer_queries(serializer)
-            if getattr(serializer, "many", False):
+            if getattr(serializer, "many", False) and isinstance(
+                serializer.instance, QuerySet
+            ):
```

Comment:
Do we need the first part of the `if` any more? Could this ever be called with `many=False` but still be given a queryset?

Also I wonder if rather than explicitly type checking this we could 🦆 type inside `fetch`:

```
def fetch(queryset):
    try:
        queryset._fetch_all()
    except AttributeError:
        pass
    return queryset
```

What dya reckon?

Reply:
I did think about modifying fetch, but really fetching a list is a bit pointless, so i'd rather go for something a bit stricter there.

In theory I think we can remove the many check, lemme check.